tests/FlexTest.php

```diff
@@ -78,6 +97,36 @@ public function getRecordTests()
         ];
     }
 
+    /**
+     * @dataProvider getLockFileTestCases
+     */
+    public function testSymfonyLock($composerEnv, $symfonyEnv, $expectedComposer, $expectedSymfony)
+    {
+        putenv("COMPOSER=" . $composerEnv);
+        putenv("SYMFONY_LOCKFILE_PATH=" . $symfonyEnv);
+
+        $flex = new Flex();
+        $symfonyLockFile = $this->invokeMethod($flex, 'getSymfonyLockFilePath', []);
+        $composerLockFile = $this->invokeMethod($flex, 'getComposerLockFilePath', []);
+        $this->assertEquals($expectedComposer, $composerLockFile);
+        $this->assertEquals($expectedSymfony, $symfonyLockFile);
+
+        putenv("COMPOSER=''");
+        putenv("SYMFONY_LOCKFILE_PATH=''");
+    }
+
+    /**
+     * dataset0: custom COMPOSER env is set, generates correct symfony lock
+     * dataset1: nothing overriden, just verify default names
+     * dataset2: composer = default, custom symfony.lock path
+     */
+    public function getLockFileTestCases()
+    {
+        return  [["composer.beta.json","", "composer.beta.lock", "symfony.beta.lock"],
+                 ["","", "./composer.lock", "./symfony.lock"],
+                 ["","symfony.beta.lock", "./composer.lock", "symfony.beta.lock"]];
+    }
```

Comment:
You can use a generator here and simply refactor this as:

```php
yield ["composer.beta.json","", "composer.beta.lock", "symfony.beta.lock"];
yield ["","", "./composer.lock", "./symfony.lock"];
yield ["","symfony.beta.lock", "./composer.lock", "symfony.beta.lock"];
```

Better readability, less memory usage :)

Reply:
I'm not sure memory usage will be that different. AFAIK, PHPUnit turns the generator into an array anyway.

But this indeed improves readability.

Reply:
fixed!